subworkflows/local/input_check.nf

```diff
@@ -35,16 +58,49 @@ def create_fastq_channel(LinkedHashMap row) {
 
     // add path(s) of the fastq file(s) to the meta map
     def fastq_meta = []
+    def fastqs = []
     if (!file(row.fastq_1).exists()) {
         exit 1, "ERROR: Please check input samplesheet -> Read 1 FastQ file does not exist!\n${row.fastq_1}"
     }
     if (meta.single_end) {
-        fastq_meta = [ meta, [ file(row.fastq_1) ] ]
+        fastqs = [ file(row.fastq_1) ]
     } else {
         if (!file(row.fastq_2).exists()) {
             exit 1, "ERROR: Please check input samplesheet -> Read 2 FastQ file does not exist!\n${row.fastq_2}"
         }
-        fastq_meta = [ meta, [ file(row.fastq_1), file(row.fastq_2) ] ]
+        fastqs = [ file(row.fastq_1), file(row.fastq_2) ]
+        if (row.sample_type == "atac") {
+            if (row.fastq_barcode == "") {
+                exit 1, "ERROR: Please check input samplesheet -> Barcode FastQ (Dual index i5 read) file is missing!\n"
+            }
+            if (!file(row.fastq_barcode).exists()) {
+                exit 1, "ERROR: Please check input samplesheet -> Barcode FastQ (Dual index i5 read) file does not exist!" +
+                        "\n${row.fastq_barcode}"
+            }
+            fastqs.add(file(row.fastq_barcode))
+        }
+    }
+
+    // define meta_data for multiome
+    def multi_meta  = []
+    multi_meta      = row.sample_type ? [row.sample_type] : [params.sample_type]
+
+    if (params.aligner == "cellrangerarc"){
+        sub_sample = row.fastq_1.split("/")[-1].replaceAll("_S[0-9]+_L[0-9]+_R1_[0-9]+.fastq.gz","")
+        fastqs.each{
+            if(!it.name.contains(sub_sample)){
+                exit 1, "ERROR: Please check input samplesheet -> Some files do not have the same sample name " +
+                        "${sub_sample} in common!\n${it}"
+            }
+        }
+        multi_meta.add(sub_sample)
     }
```

Comment:
What's the expected structure of `multi_meta`? Couldn't `sample_type` and `sub_sample` just be added as keys to the regular meta, i.e. `meta.sample_type`, `meta.sub_sample`?

Reply:
To clarify here is an example samplesheet and the lib.csv that is generated and needed for cellranger-arc

sample sheet:
sample,fastq_1,fastq_2,fastq_barcode,sample_type
test_scARC,test_scARC_atac_S1_L001_R1_001.fastq.gz,test_scARC_atac_S1_L001_R2_001.fastq.gz,test_scARC_atac_S1_L001_I2_001.fastq.gz,atac
test_scARC,test_scARC_gex_S1_L001_R1_001.fastq.gz,test_scARC_gex_S1_L001_R2_001.fastq.gz,,gex

fastqs,sample,library_type
path/fastqs,test_scARC_atac,Chromatin Accessibility
path/fastqs,test_scARC_gex,Gene Expression

Thus the inherent problem is here that your sample consist of two subsamples (scATAC and scRNA/GEX). Both subsamples can have several fastqs. Furthermore, those subsample names can differ therefore the test_scARC_atac and test_scARC_gex names. 

To answer now your question for this particular example multi_meta looks like this:
[atac, test_scARC_atac, gex, test_scARC_gex]

You are right I probably could have just added those two additional keys.

Should I change it like that?

Reply:
I split the multi_meta into the recommeded two keys. It is easier to handle and more intuitive.